tests/e2e/util/kubectl/kubectl.go

```diff
@@ -41,8 +49,8 @@ func kubectl(format string, args ...interface{}) string {
 }
 
 // CreateFromString creates a resource from the given yaml string
-func CreateFromString(yamlString string) error {
-	cmd := kubectl("create -f -")
+func CreateFromString(yamlString string, kubeconfig ...string) error {
```

Comment:
Hmm, not sure adding `kubeconfig` to all the functions is a good approach. Instead, we should be able to create  a new kubectl object and pass the kubeconfig to it.

Reply:
But now I'm also thinking that we should maybe get rid of this `kubectl` thing altogether and just invoke `shell.ExecuteCommand` directly. Wouldn't that make the code simpler? Currently, `kubectl.go` represents another abstraction layer that is starting to become hard to understand due to all the different usage options.

Reply:
Or maybe use the builder pattern to construct the command.

Reply:
> But now I'm also thinking that we should maybe get rid of this `kubectl` thing altogether and just invoke `shell.ExecuteCommand` directly. Wouldn't that make the code simpler? Currently, `kubectl.go` represents another abstraction layer that is starting to become hard to understand due to all the different usage options.

Yes, using `shell.ExecuteCommand` will be simpler to get rid of the kubeconfig param, but I think it will not be so useful because we will need to set everything in the steps of the test, and that will make the code less readable as is now. I can try the builder pattern, which I think can be a better

Reply:
Hey @luksa I marked the PR as ready for review, but I did not address the `kubectl` handle kubeconfig on this PR. I'll do it in a consecutive PR, I think the PR is complex enough to add more complexity by refactoring the `kubectl` util